src/ng/directive/input.js

```diff
@@ -1456,7 +1456,15 @@ function createDateParser(regexp, mapping) {
             map[mapping[index]] = +part;
           }
         });
-        return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+
+        var date = new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+        if (map.yyyy < 100) {
+          // In the constructor, 2-digit years map to 1900-1999.
+          // Use `setFullYear()` to set the correct year.
+          date.setFullYear(map.yyyy);
```

Comment:
How about putting this before the `new Date` and doing `map.yyyy += 1900`? Then the `return new Date(...)` can be left as-is and maybe the `date` => `previousDate` change isn't needed?

Reply:
But we don't want the year to be +1900 :grin:
(Also, the `date` --> `previousDate` wasn't really necessary, but I thought `previousDate` conveys the nature of the argument better :grin: Happy to change it back.)

Reply:
> But we don't want the year to be +1900 😁

Right, that's basically what we are preventing...

I'm fine keeping the `previousDate` change if the new variable is needed, which it appears it is (unlike what I first thought).